### PR TITLE
added the setup

### PR DESCRIPTION
**Pull request by Tresor11**

### CODE REVIEW

**Comment by WCanirinka** [comment]
We are submitting for review

File: file.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: false
+
+# first method bubble sort
+
+def bubble_sort(arr)
+  arr.sort!
+end
```
Comment (wathigo) [inline file.rb]: **[REQUIRED]**
* We should not use `#sort` method to implement bubble sort algorithm.
* We must implement the correct methodology for the bubble sort algorithm. Check out [this](https://www.geeksforgeeks.org/bubble-sort/). Check this too :point_down: 
rec

File: file.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# second method
+
+def bubble_sort_by(arr)
+  arr.sort! do |x, y|
+    x.length - y.length
+  end
+end
```
Comment (wathigo) [inline file.rb]: **[REQUIRED]**
* We Should not use inbuilt `#sort` method.
* Remember to raise an error in no block is given.

**Review by wathigo — CHANGES_REQUESTED** [review]
## Changes Required 

Let's fix the issues pointed out in the comments :point_down:  

**After implementing the changes, submit another request for review**

_Happy Coding :computer:  
Reviewed by: **Simon Wathigo**_

**Comment by WCanirinka** [comment]
Requesting for a code review after the correcting the required changes

File: file.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+# bubble_sort([4, 3, 78, 2, 0, 2])
+
+def bubble_sort_by(array)
+  sorted = true
+  while sorted
+    sorted = false
+    (arr.length - 1).times do |a|
+      if yield(array[a], array[a + 1]).positive?
```
Comment (stratospherique) [inline file.rb]: - Calling this method raises the following error : `undefined local variable or method 'arr'`. :pray: fix it.

File: file.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    sorted = false
+    (array.length - 1).times do |a|
+      if array[a + 1] < array[a]
+        array[a], array[a + 1] = array[a + 1], array[a]
+        sorted = true
+      end
+    end
+  end
+  print array
+end
```
Comment (stratospherique) [inline file.rb]: - Calling bubble_sort method as below returns nil. :pray: fix it
  ```ruby
   arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
   p bubble_sort(arr) # returns nil
  ```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great progress Team :clap: :clap: However, further changes needs to be made in order for us to approve your project. check :arrow_down: 

- **[REQUIRED]** Check the code comments below, and use the notes to refactor your methods.

- **[REQUIRED]** No debugging code (i.e remove commented code in your case)

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
_**Ahmed Mahfoudh**_

**Review by ebukaume — APPROVED** [review]
Hi @WCanirinka :wave: 

**Nice job! You can go ahead and merge this branch.**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Pull request merged**